### ISSUE TITLE
Stop code

### ISSUE THREAD

**Issue by hypersniper05** [bublint/ue5-llama-lora#4]
After days of getting a similar dataset to work, I've found it responses best if you add a stop code at the end of the sentences or line in your dataset. I used `</S>`. Then you add this in the parameters tab. Otherwise the model will keep quoting  the text. Perhaps mine was overfitted.

**Comment by TianweiXing** [comment]
@hypersniper05 May I ask a quick question: if I understand correctly, the model is fine-tuning on the dataset stored in the .txt file. Does it treat each valid line as a sample, regardless of the length/content of the line? (without the format of instruction/input/output)
Thanks!

**Comment by umairafzal9245** [comment]
@hypersniper05 can you please tell me your hyperparameters for training?

**Comment by hypersniper05** [comment]
@umairafzal9245 sure, there's been recent updates to text generation webui and you no longer need the stop code. Can you tell me more or less what type of data are you training ? You don't need to train much . I've found 1 epoch is enough. Default settings in text generation webui is good. If you train in raw data the model will mimic it so you might want to train on question answer pair in alpaca format (or have chatgpt code interpreter fix the jsoj for you)

**Comment by umairafzal9245** [comment]
@hypersniper05 i want to teach the model about different people i am using rank=512, alpha=1024, llama7B as base model but the model is generating random words e.g if i ask it about the education of fatima zafar, its output is completed garbage. samples are given below

fatima zafar:

fatima zafar is currently working as a technical consultant at peregrine corporation

peregrine corporation operates in different industries fuel retail and distribution, retail operations, property development, wholesale & supply chain

fatima zafar has been working at peregrine corporation since 5-2019

fatima zafar has a total experience of 9 years in various industries information technology & services, business intelligence, cloud services, business application development, information security, quality assurance, enterprise resource planning, business process outsourcing, digital commerce, application integration & modernization, user experience, staff augmentation, system integration

fatima zafar is a highly skilled professional based in greater adelaide area, australia

fatima zafar has a variety of skills including asp.net, c++, powerpoint, object-oriented programming (oop), css, x++, business process, teamwork, oop, visual studio, customer service, microsoft excel, linq, enterprise resource planning (erp), javascript, sql server reporting services (ssrs), public speaking, microsoft dynamics ax, c, software development, blogging, microsoft office, business analysis, photoshop, social media, microsoft sql server, research, php, databases, microsoft word, requirements analysis, sql and vb.net

fatima zafar education:

fatima zafar completed his bachelor of science (b.s.) in computer science from nuces-fast in year 2009 to 2013

fatima zafar experience:

fatima zafar worked as a technical consultant at spendless shoes pty ltd from 11-2018 to 5-2019

fatima zafar worked as a technical consultant at empired ltd from 6-2017 to 10-2018

fatima zafar worked as a senior software engineer at systems limited from 5-2015 to 1-2017

**Comment by hypersniper05** [comment]
@umairafzal9245 Don't fine-tune, the model is able to reply already you just feed it a prompt and the document. Look into vector search with similarity cosine. Basically, you give the model a prompt like:

 "System:
 You are to give an accurate and detailed personal bibliography of the person below using the provided documents. Take into account the user's question when answering. If you cannot answer the question with the provided documents reply with a clarification response for the user. 
 User: 
 Who is Fatima Zafar?
 Documents: 
 fatima zafar is currently working as a technical consultant at peregrine corporation peregrine corporation operates in different industries fuel retail and distribution, retail operations, property development, wholesale & supply chain
fatima zafar has been working at peregrine corporation since 5-2019
fatima zafar has a total experience of 9 years in various industries information technology & services, business intelligence, cloud services, business application development, information security, quality assurance, enterprise resource planning, business process outsourcing, digital commerce, application integration & modernization, user experience, staff augmentation, system integration
fatima zafar is a highly skilled professional based in greater adelaide area, australia
fatima zafar has a variety of skills including asp.net, c++, powerpoint, object-oriented programming (oop), css, x++, business process, teamwork, oop, visual studio, customer service, microsoft excel, linq, enterprise resource planning (erp), javascript, sql server reporting services (ssrs), public speaking, microsoft dynamics ax, c, software development, blogging, microsoft office, business analysis, photoshop, social media, microsoft sql server, research, php, databases, microsoft word, requirements analysis, sql and vb.net
fatima zafar education:
fatima zafar completed his bachelor of science (b.s.) in computer science from nuces-fast in year 2009 to 2013
fatima zafar experience:
fatima zafar worked as a technical consultant at spendless shoes pty ltd from 11-2018 to 5-2019
fatima zafar worked as a technical consultant at empired ltd from 6-2017 to 10-2018
fatima zafar worked as a senior software engineer at systems limited from 5-2015 to 1-2017"


Output:
"Fatima Zafar is a technical consultant currently working at Peregrine Corporation, which operates in different industries such as fuel retail and distribution, retail operations, property development, wholesale & supply chain. She has been working at Peregrine Corporation since May 2019 and has a total experience of 9 years in various industries including information technology & services, business intelligence, cloud services, business application development, information security, quality assurance, enterprise resource planning, business process outsourcing, digital commerce, application integration & modernization, user experience, staff augmentation, system integration. Fatima Zafar is highly skilled and based in the Greater Adelaide area, Australia. She has a variety of skills including ASP.NET, C++, PowerPoint, Object-Oriented Programming (OOP), CSS, X++, Business Process, Teamwork, OOP, Visual Studio, Customer Service, Microsoft Excel, LINQ, Enterprise Resource Planning (ERP), JavaScript, SQL Server Reporting Services (SSRS), Public Speaking, Microsoft Dynamics AX, C, Software Development, Blogging, Microsoft Office, Business Analysis, Photoshop, Social Media, Microsoft SQL Server, Research, PHP, Databases, Microsoft Word, Requirements Analysis, SQL and VB.NET. Fatima Zafar completed her Bachelor of Science (B.S.) in Computer Science from Nuces-Fast in the years 2009 to 2013. Her work experience includes working as a Technical Consultant at Spendless Shoes Pty Ltd from November 2018 to May 2019, Technical Consultant at Empireired Ltd from June 2017 to October 2018, and Senior Software Engineer at Systems Limited from May 2015 to January 2017."


Here are is the model: [https://huggingface.co/lmsys/vicuna-7b-v1.5-16k](https://huggingface.co/lmsys/vicuna-7b-v1.5-16k)

Here are the params I used:
<img width="1259" alt="params" src="https://github.com/bublint/ue5-llama-lora/assets/29990615/a320449c-c82b-4c44-a1ea-7ae101fd943f">
<img width="1269" alt="params 2" src="https://github.com/bublint/ue5-llama-lora/assets/29990615/2d9549b4-f3ef-490a-8a71-435a07ac29b6">
<img width="725" alt="param3" src="https://github.com/bublint/ue5-llama-lora/assets/29990615/b3d8a9bc-0daf-4feb-88fc-f74c74b0310f">
<img width="688" alt="output" src="https://github.com/bublint/ue5-llama-lora/assets/29990615/8084b19f-17c2-4211-a5c6-03e766bf9522">

**Comment by umairafzal9245** [comment]
@hypersniper05 i want to achieve this using finetunning so that i dont have to provide the context to model at runtime. like it already knows a lot about entities like elon musk etc, although we dont provide it with information of elon musk at real time.

**Comment by hypersniper05** [comment]
It does not work that way with raw text. I think you are misunderstanding the fine-tuning stage. You are teaching it how to reply. If you must do this you will need to bring the Lora rank way lower like 16 . If you go too high you train too deep and influence the model too much and it will just put out what you trained it on. You'll have to play with the epoch, learning rate, Lora rank, and alpha to get it just right and also monitor the loss doesn't get too low 1.5 is pretty good.  The best thing to do is to try to convert your data into question-and-answer pairs. You can use a LLM to help you.
